querybook/server/lib/scheduled_datadoc/legacy.py

```diff
@@ -8,6 +11,28 @@ def set_key_if_exists(to_dict: Dict, from_dict: Dict, key: str):
         to_dict[key] = from_dict[key]
 
 
+def is_legacy_notification_format(schedule_config):
+    return (
+        schedule_config.get("notify_with")
+        and schedule_config.get("notify_on")
+        and not schedule_config.get("notifications")
+    )
+
+
+def convert_legacy_notification_to_new_format(schedule_config):
+    user = user_logic.get_user_by_id(schedule_config.get("user_id"))
+    notify_to = (
+        user.email if schedule_config.get("notify_with") == "email" else user.username
+    )
+    return [
+        {
+            "on": schedule_config.get("notify_on"),
+            "with": schedule_config.get("notify_with"),
+            "config": {"to": [notify_to]},
+        }
+    ]
+
+
 def convert_if_legacy_datadoc_schedule(schedule_config: Dict) -> Dict:
```

Comment:
A feedback from @czgu
> can we have separate convert function from schedule format v0(single export) to v1(current notification format) and to v2(new notification format), so that we can track the schedule format change history.
we can keep the function of `convert_if_legacy_datadoc_schedule` as it will be the external interface be used by the `run_data_doc` and `get_datadoc_schedule` api, but all the above two convert function in it.